src/PageContent/RequestPickup/RecurringPickupRequest.js

```diff
@@ -0,0 +1,284 @@
+            if (!dgUid) dgUid = null;
+
+            var durationValue = null;
+            if (event.target.durationType.value === RequestDurationType.DATE) {
+                durationValue = event.target.endDate.value;
+            } else {
+                durationValue = event.target.numRecurrences.value;
+            }
+
+            // create DeliveryRequest object
+            var deliveryRequest = {
+                startDate: event.target.startDate.value,
+                duration:{
+                    type: event.target.durationType.value,
+                    value: durationValue
+                },
+                repeats: event.target.repeats.value,
+                startTime: event.target.startTime.value,
+                endTime: event.target.endTime.value,
+                primaryContact: event.target.primaryContact.value,
+                notes: event.target.notes.value,
+                receiveingAgency: {
+                    uid: raUid,
+                    confirmed: false
+                },
+                delivererGroup: {
+                    uid: dgUid,
+                    confirmed: false
+                },
+                requestTimestamp: Date.now()
+            };
+            let date = new Date(deliveryRequest.startDate);
+            let day = date.getDay();
+            let days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
+
+            this.setState({
+                formInfo: deliveryRequest,
+                dayOfWeek: days[day]
+            });
+            
+            this.toggleModal();
+        }
+    }
+
+    render() {
+        return (
+            <div className="form" id="recurringForm">
+                <form onSubmit={this.createRequest}>
+                    <div className="info">
+                        <p id="form-heading">Schedule Recurring Pickup</p>
+                        <p className="error">{this.state.errors['endBeforeStart']}</p>
+                        <p className="error">{this.state.errors['startTime']}</p>
+                        <p className="error">{this.state.errors['endTime']}</p>
+                        <p className="error">{this.state.errors['recurTimes']}</p>
+                        <p className="error">{this.state.errors['startDate']}</p>
+                        <p className="error">{this.state.errors['time']}</p>
+                        <span className="flex">
+                            <span className="grid">
+                                <label>Start Date  <span className="red">*</span></label><br/>
+                                <input type="date" name="startDate" onChange={this.handleChange.bind(this, 'startDate')} required/><br/>
+                            </span>
+                            <span className="grid">
+                                <label className="container smaller">
+                                    <input type="radio" name="durationType" value={RequestDurationType.RECUR} required/>
```

Comment:
add `handleChange` for `durationType` so that you can check against it like mentions above in validation